.circleci/config.yml

```diff
@@ -49,6 +49,35 @@ commands:
       - run: git branch -f stable HEAD
       - run: git push origin stable
 
+  build_sphinx_documentation:
+    description: Build opera documentation with Sphinx
+    steps:
+      - common_setup
+      - checkout
+      - run: make docs
+      - persist_to_workspace:
+          root: docs/_build
+          paths: .
+
+  deploy_docs_to_github_pages:
+    description: Build Sphinx documentation and push GitHub stable branch which points to the latest relased tag/commit
+    steps:
+      - add_ssh_keys:
+          fingerprints:
+            - "b6:94:61:c3:e1:8f:25:e5:43:6d:51:9e:43:c4:16:11"
+      - checkout
+      - run: git checkout --orphan gh-pages
+      - run: git rm --cached -r .
+      - run: rm -rf *
+      - attach_workspace:
+          at: .
+      - run: touch .nojekyll
+      - run: git config --global user.email circleci@circleci
+      - run: git config --global user.name CircleCI
+      - run: git add *
+      - run: git commit -m "Add generated opera docs for GitHub pages"
+      - run: git push origin gh-pages
+
```

Comment:
Doing the doc build in tow stages feels rather awkward. Maybe we can do it in a single step by utilizing git worktrees? This is how one would do that manually (translating this to the CircleCI YAML is left as an exercise for the reader ;)):

```
make docs
git worktree add gh-pages gh-pages
cd gh-pages
git rm -rf . )
cp -r ../docs/_build/* .
git add .
git comm -m "..."
git push origin gh-pages
```

Reply:
Thanks for the comment and suggestion, as a reader I tried to stick to your commands and translated them to CircleCI YAML entries.